File: 프로그래머스/1/12982. 예산/예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        int count = 0;
        Arrays.sort(d);
        for (int i : d) {
            if (budget >= i) {
                budget -= i;
                count++;
            }
            if (budget <= 0) {
                break;
            }
        }
        return count;
    }
}

File: 프로그래머스/1/12982. 예산/예산.swift
import Foundation

func solution(_ d:[Int], _ budget:Int) -> Int {
    var budget = budget
    var count = 0
    for i in d.sorted(by: <) {
        if budget >= i {
            budget -= i
            count += 1
        }
        if budget <= 0 {
            break
        }
    }
    return count
}